src/core/init/utils/media_source_duration_updater.ts

```diff
@@ -192,9 +192,9 @@ function setMediaSourceDuration(
   } else {
     const oldDuration = mediaSource.duration;
     try {
-      log.info("Init: Updating duration", newDuration);
-      mediaSource.duration = newDuration;
       if (mediaSource.readyState === "open" && !isFinite(newDuration)) {
+        log.info("Init: Updating duration", newDuration);
+        mediaSource.duration = newDuration;
```

Comment:
Are we sure about this? There may be an issue for perhaps devices where we would have set duration to `Infinity`?